pkg/stanza/docs/operators/journald_input.md

```diff
@@ -33,7 +35,65 @@ The `journald_input` operator will use the `__REALTIME_TIMESTAMP` field of the j
 - type: journald_input
   priority: emerg..err
 ```
-#### Simple journald input
+
+#### Matches
+
+The following configuration:
+
+```yaml
+- type: journald_input
+  matches:
+    - _SYSTEMD_UNIT: ssh
+    - _SYSTEMD_UNIT: kubelet
+      _UID: "1000"
+```
+
+will be passed to `journalctl` as the following arguments: `journalctl ... _SYSTEMD_UNIT=ssh + _SYSTEMD_UNIT=kubelet _UID=1000`,
+which is going to retrieve all entries which match at least one of the following rules:
+
+- `_SYSTEMD_UNIT` is `ssh`
+- `_SYSTEMD_UNIT` is `kubelet` and `_UID` is `1000`
+
+#### Multiple filtering options
+
+In case of using multiple following options, conditions between them are logically `AND`ed and within them are logically `OR`ed:
+
+```text
+( priority )
+AND
+( units[0] OR units[1] OR units[2] OR ... units[U] )
+AND
+( matches[0] OR matches[1] OR matches[2] OR ... matches[M] )
+```
+
+Consider the following example:
+
+```yaml
+- type: journald_input
+  matches:
+    - _SYSTEMD_UNIT: ssh
+    - _SYSTEMD_UNIT: kubelet
+      _UID: "1000"
+  units:
+    - kubelet
+    - systemd
+  priority: info
+```
+
+The above configuration will be passed to `journalctl` as the following arguments
+`journalctl ... --priority=info --unit=kubelet --unit=systemd _SYSTEMD_UNIT=ssh + _SYSTEMD_UNIT=kubelet _UID=1000`,
+which is going to effectively retrieve all entries which matches the following set of rules:
+
+- `_PRIORITY` is `6`, and
+- `_SYSTEMD_UNIT` is `kubelet` or `systemd`, and
+- entry matches at least one of the following rules:
+
+  - `_SYSTEMD_UNIT` is `ssh`
+  - `_SYSTEMD_UNIT` is `kubelet` and `_UID` is `1000`
+
+Note, that if you use some fields which aren't associated with an entry, the entry will always be filtered out.
```

Comment:
Is it worth noting that this configuration will not emit ssh or systemd logs because of the `AND` between matches and units?

Reply:
Added additional note to explicitly say that the example configuration won't get any entries for `ssh` and `systemd`